E2E: QoL updates to package scripts and update documentation.

worldomonation opened the pull request:
#### Proposed Changes

This PR arose as a result of the conversation [here](p1671038940457179-slack-CQD1HH4MA).

Key changes:
- remove existing decrypted secrets file prior to decrypting the new one, to prevent errors.
- add package scripts in `test/e2e` to make it easier to launch a debug instance of Playwright.
- short circuit the cookie saving mechanism if PWDEBUG mode is enabled.
- update `yarn` commands in the documentation.

#### Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)
  - [x] Code Style

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

worldomonation committed:
packages/calypso-e2e/package.json
- when decrypting e2e secrets using `yarn decrypt-secrets`, remove the existing secrets file.

worldomonation committed:
test/e2e/package.json
- add `start` and `start:debug` methods to simplify the process of starting a new debug instance.

b/packages/calypso-e2e/src/jest-playwright-config/global-setup.ts
- set timeout for the cookie refresh process.
- short circuit if the PWDEBUG mode is set.

worldomonation committed:
packages/calypso-e2e/package.json
- use ; instead of &

worldomonation commented on packages/calypso-e2e/package.json:
```diff
@@ -43,6 +43,6 @@
 		"build": "tsc --build ./tsconfig.json",
 		"encrypt-secrets": "openssl enc -md sha1 -aes-256-cbc -pass env:E2E_SECRETS_KEY -out ./src/secrets/encrypted.enc -in ./src/secrets/decrypted-secrets.json",
-		"decrypt-secrets": "openssl enc -md sha1 -aes-256-cbc -d -pass env:E2E_SECRETS_KEY -in ./src/secrets/encrypted.enc -out ./src/secrets/decrypted-secrets.json"
+		"decrypt-secrets": "rm ./src/secrets/decrypted-secrets.json; openssl enc -md sha1 -aes-256-cbc -d -pass env:E2E_SECRETS_KEY -in ./src/secrets/encrypted.enc -out ./src/secrets/decrypted-secrets.json"
```
This is often the source of headache for folks - you'd think the decryption process would overwrite the existing file "cleanly", but it ends up causing problems.

karenroldan replied:
Thank you so much for this improvement! 🚀 🎉

worldomonation commented on packages/calypso-e2e/src/jest-playwright-config/global-setup.ts:
```diff
@@ -7,6 +7,12 @@ import pwConfig from './playwright-config';
 	const { AUTHENTICATE_ACCOUNTS } = envVariables;
 
+	// If PWDEBUG mode is enabled (stepping through each step)
+	// don't execute the cookie refresh.
+	if ( process.env.PWDEBUG ) {
+		return;
+	}
```
I'm open to discussion on this one.

My line of thinking was that:
- If I'm using PWDEBUG mode to step through a script
- then I probably do not need or want to deal with the three cookie saving prompts

worldomonation commented on test/e2e/package.json:
```diff
@@ -18,7 +18,9 @@
 		"clean:output": "rm -rf ./allure-results && rm -rf ./results",
-		"build": "yarn workspace @automattic/jest-circus-allure-reporter build && yarn workspace @automattic/calypso-e2e build"
+		"build": "yarn workspace @automattic/jest-circus-allure-reporter build && yarn workspace @automattic/calypso-e2e build",
+		"start": "yarn jest",
+		"start:debug": "PWDEBUG=1 DEBUG=pw:api yarn run start"
```
`yarn start <spec_path` doesn't actually save any keystrokes, I just realized.

Perhaps only `yarn debug` is necessary for now?

karenroldan replied:
**thought:** I prefer `yarn start <spec_path>` despite more keystrokes because it's "easier" to remember what it tries to do. 

`yarn debug` is also preferable. 👍🏻

worldomonation replied:
I was thinking in bed before sleeping and think maybe `yarn test` is also a viable alternative.

`yarn test` and either `yarn test:debug` or `yarn debug`.

worldomonation replied:
Changed to `yarn test` and `yarn debug` in b6b9f48696703a823482b4b03d505bcebeb85399.

worldomonation committed:
test/e2e/docs/
- update docs with new package commands

test/e2e/package.json
- change commands to `test` and `debug`
- add mobile variant of each.

worldomonation commented:
Also updated the page in FG: PCYsg-vnR-p2

karenroldan reviewed (approved):
Looks good to me! Thank you for the improvements 🚀 🎉
